store/view.go

```diff
@@ -104,6 +115,25 @@ func (s *Store) DeleteView(ctx context.Context, delete *api.ViewDelete) error {
 //
 // private functions
 //
+func generateViewActions(oldViewMap map[string]*viewRaw, newViewMap map[string]*api.ViewCreate) ([]*api.ViewDelete, []*api.ViewCreate) {
+	var deletes []*api.ViewDelete
+	var creates []*api.ViewCreate
+	for k, oldValue := range oldViewMap {
+		newValue, ok := newViewMap[k]
+		if !ok {
+			deletes = append(deletes, &api.ViewDelete{ID: oldValue.ID})
+		} else if ok && (oldValue.Definition != newValue.Definition || oldValue.Comment != newValue.Comment) {
```

Comment:
What if it's just a rename https://www.postgresql.org/docs/14/sql-alterview.html

Reply:
Yeah, we are trying to minimize the writes to databases. Renames or updates are rare, and we don't need to fully optimize these cases for simplicity.

Reply:
Since it's rare, then it doesn't matter much that we cover it here if we want to minimize the writes.

I am thinking of completeness & readability, since Name is the only leftover field and readers may wonder why we explicitly ignore its change.

Reply:
From our perspective, ALTER VIEW RENAME TO can be considered as the same as a recreation of the view. This implementation is easier than doing a patch (database row update). By intuition, people may think the definition of a view can change while the name does not. However, this is different from database perspective.

Reply:
NM. I misread, I thought we didn't cover the "ALTER VIEW RENAME TO" case at all